src/expr/src/scalar/func/impls/mz_timestamp.rs

```diff
@@ -128,6 +129,18 @@ sqlfunc!(
     }
 );
 
+sqlfunc!(
+    #[sqlname = "date_to_mz_timestamp"]
+    #[is_monotone = true]
+    fn cast_date_to_mz_timestamp(a: Date) -> Result<Timestamp, EvalError> {
```

Comment:
Ah good call, we should set that here, since the precision of `date` is less than `mz_timestamp`.

For `cast_timestamp_to_mz_timestamp`, Maddy explicitly didn't set `#[preserves_uniqueness = true]` in the PR that added all of the initial implicit casts, while she did set it on the the others like `cast_int32_to_mz_timestamp`. I think that's because `timestamp` can have a precision of microseconds, while `mz_timestamp` has a precision of milliseconds.

Reply:
Oh, I see, ok!